addons/addon-base-raas/packages/base-raas-services/lib/study/__tests__/study-permission-service.test.js

```diff
@@ -12,924 +12,575 @@
+jest.mock('../../user/user-service');
+
+const Aws = require('@aws-ee/base-services/lib/aws/aws-service');
+const Logger = require('@aws-ee/base-services/lib/logger/logger-service');
+const LockService = require('@aws-ee/base-services/lib/lock/lock-service');
+const DbService = require('@aws-ee/base-services/lib/db-service');
+const PluginRegistryService = require('@aws-ee/base-services/lib/plugin-registry/plugin-registry-service');
+const SettingsService = require('@aws-ee/base-services/lib/settings/env-settings-service');
+const AuthService = require('@aws-ee/base-services/lib/authorization/authorization-service');
+const AuditService = require('@aws-ee/base-services/lib/audit/audit-writer-service');
+const JsonSchemaValidationService = require('@aws-ee/base-services/lib/json-schema-validation-service');
+const UserService = require('../../user/user-service');
+const StudyPermissionService = require('../study-permission-service');
 
-jest.mock('@aws-ee/base-services/lib/user/user-service');
-const UserServiceMock = require('@aws-ee/base-services/lib/user/user-service');
+const { getEmptyStudyPermissions } = require('../helpers/entities/study-permissions-methods');
+const { getEmptyUserPermissions } = require('../helpers/entities/user-permissions-methods');
 
-const StudyPermissionService = require('../study-permission-service');
+describe('StudyPermissionService', () => {
+  let service;
+  let dbService;
 
-// Tested functions: create, update, delete
-describe('studyPermissionService', () => {
-  let service = null;
-  let dbService = null;
-  let lockService = null;
-  const error = { code: 'ConditionalCheckFailedException' };
   beforeEach(async () => {
+    // Initialize services container and register dependencies
     const container = new ServicesContainer();
+
+    container.register('dbService', new DbService());
     container.register('jsonSchemaValidationService', new JsonSchemaValidationService());
-    container.register('dbService', new DbServiceMock());
-    container.register('auditWriterService', new AuditServiceMock());
-    container.register('settings', new SettingsServiceMock());
-    container.register('lockService', new LockServiceMock());
-    container.register('userService', new UserServiceMock());
+    container.register('settings', new SettingsService());
+    container.register('aws', new Aws());
+    container.register('log', new Logger());
+    container.register('lockService', new LockService());
+    container.register('userService', new UserService());
+    container.register('pluginRegistryService', new PluginRegistryService());
+    container.register('authorizationService', new AuthService());
+    container.register('auditWriterService', new AuditService());
     container.register('studyPermissionService', new StudyPermissionService());
-
     await container.initServices();
 
     service = await container.find('studyPermissionService');
```

Comment:
Is there a test covering the part where a readOnly study cannot have readWrite or writeOnly users. Perhaps I missed it.

Reply:
Yes, I have many tests to cover this case in addons/addon-base-raas/packages/base-raas-services/lib/study/helpers/entities/__tests__/study-methods.test.js